great_expectations/expectations/expectation.py

```diff
@@ -1726,39 +1739,63 @@ def _validate_metric_value_between(
 
             if min_value is not None:
                 try:
-                    min_value = parse(min_value)
+                    min_value_for_comparison = round(
+                        (parse(min_value) - EPOCH_DATE_TIME).total_seconds()
+                    )
+                    metric_value_for_comparison = round(
+                        (metric_value - EPOCH_DATE_TIME).total_seconds()
+                    )
                 except TypeError:
                     pass
 
             if max_value is not None:
                 try:
-                    max_value = parse(max_value)
+                    max_value_for_comparison = round(
+                        (parse(max_value) - EPOCH_DATE_TIME).total_seconds()
+                    )
+                    metric_value_for_comparison = round(
+                        (metric_value - EPOCH_DATE_TIME).total_seconds()
+                    )
                 except TypeError:
                     pass
 
-        elif isinstance(metric_value, datetime.datetime):
-            if isinstance(min_value, str):
-                min_value = parse(min_value)
-            if isinstance(max_value, str):
-                max_value = parse(max_value)
-
         if not isinstance(metric_value, datetime.datetime) and pd.isnull(metric_value):
             return {"success": False, "result": {"observed_value": None}}
 
+        if isinstance(metric_value, datetime.datetime):
+            metric_value_for_comparison = round(
+                (metric_value - EPOCH_DATE_TIME).total_seconds()
+            )
+            if isinstance(min_value, str):
+                try:
+                    min_value_for_comparison = round(
+                        (parse(min_value) - EPOCH_DATE_TIME).total_seconds()
+                    )
+                except TypeError:
+                    pass
+
+            if isinstance(max_value, str):
+                try:
+                    max_value_for_comparison = round(
+                        (parse(max_value) - EPOCH_DATE_TIME).total_seconds()
+                    )
+                except TypeError:
+                    pass
```

Comment:
I really, really, object to rounding here.

Reply:
@jcampbell I agree.  Tests are running locally with this logic removed.

Reply:
@jcampbell Tests are now passing with only the `sqlite` provisions introduced to the `JSON` test case specifications.  I also added error handling in case the new parameter parsing from string to datetime fails.  Thanks!